Sources/TSCBasic/Path.swift

```diff
@@ -179,24 +255,50 @@ public struct AbsolutePath: Hashable {
     /// NOTE: We may want to consider adding operators such as `+` for appending
     ///       a path component.
 
-    /// NOTE: We will want to add a method to return the lowest common ancestor
-    ///       path.
-
-    /// Root directory (whose string representation is just a path separator).
-    public static let root = AbsolutePath(PathImpl.root)
+    /// Returns the lowest common ancestor path.
+    public func lowestCommonAncestor(with path: AbsolutePath) -> AbsolutePath? {
```

Comment:
Common ancestor queries is also a good candidate for FilePath. We had operations like `lexicallyRelative(to:)` in earlier pitches.